File: leetcode/c_programming/longest_substring.c
int lengthOfLongestSubstring(char* s) {
    int i, end, start, max_len, current_len;
    int charIndex[128];

    i = end = start = max_len = current_len = 0;
    for (; i < 128; i++) {
        charIndex[i] = -1;
    }

    while (s[end] != '\0') {
        if (charIndex[s[end]] != -1 && charIndex[s[end]] >= start) {
            start = charIndex[s[end]] + 1;
        }
        charIndex[s[end]] = end;

        current_len = (end - start) + 1;
        if (current_len > max_len) {
            max_len = current_len;
        }
        end++;
    }

    return max_len;
}

File: leetcode/python/longest_substring.py
#!/usr/bin/python3
"""Finds the longest substring in a given string
withoit repeating characters"""

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_idx_map = {}
        _max = 0
        start = 0

        for end in range(len(s)):
            if s[end] in char_idx_map and char_idx_map[s[end]] >= start:
                start = char_idx_map[s[end]] + 1

            char_idx_map[s[end]] = end

            _max = max(_max, (end - start) + 1 )

        return _max
